### ISSUE TITLE
Resolved: 84 pull-requests inadvertently and mistakenly closed!

### ISSUE THREAD

**Issue by abernix** [apollographql/apollo-server#4304]
### ~🚨 Repo PR Alert! 🚨~ (Resolved!)

~_(Read: Whoopsie, sorry — Please stand by!)_~

A total of 84 PRs were **inadvertently closed** by an automated process which was expected to re-target them to a new default branch (`main`, via https://github.com/apollographql/apollo-server/pull/4302), but instead re-targeted merely 23 of the open PRs and instead _closed_ 84 others.  (An additional 14 were not affected but had already been targeted to the `main` branch.)

The expectation was that the [pull-request re-targeting feature](https://github.blog/changelog/2020-05-19-pull-request-retargeting/) would handle this when https://github.com/apollographql/apollo-server/pull/4302 merged, but it seems it only partially succeeded (this is the automatic process referred to above).  I did test this on another repository prior to doing this, so I was certainly not expecting this result!

~We have contacted GitHub Support to see if they can re-open these programmatically, otherwise we will investigate further and take action ourselves.  So very sorry for PRs that were automatically closed!  I'm going to leave the repository as is for the time being, but _stand by_ for updates!
~
😭

**Comment by andrewmcgivery** [comment]
Ooooooooof. Sending good vibes lol

**Comment by abernix** [comment]
✨ Hopefully all back to normal!  There were [5 PRs](https://github.com/apollographql/apollo-server/pulls?q=is%3Apr+sort%3Aupdated-desc+closed%3A2020-06-24T21%3A10%3A00%2B03%3A00..2020-06-24T21%3A20%3A00%2B03%3A00+is%3Aclosed+) that couldn't be re-opened either because:

1. They had already been re-opened.
2. Their source branches were already deleted.

I have manually fixed all of them, or just maintained the already-re-opened PRs.

For posterity's sake:

1. <details>	
	<summary>Here's the JSON output of the affected PRs, in case this needs to be re-visited (click to expand)</summary>	
	
	```json
	{
	  "data": {
	    "prSearch": {
	      "issueCount": 84,
	      "edges": [
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDM4NTIxNjAw",
	            "number": 4298,
	            "title": "Fix (#4283 #3582): Union with interfaces issue",
	            "url": "https://github.com/apollographql/apollo-server/pull/4298",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDM3MjkxNzQ5",
	            "number": 4288,
	            "title": "Add ability to have user-specified cache headers and values",
	            "url": "https://github.com/apollographql/apollo-server/pull/4288",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDM2ODM1MjQ2",
	            "number": 4285,
	            "title": "Add app-engine deployment docs",
	            "url": "https://github.com/apollographql/apollo-server/pull/4285",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDM2MDM2MDI5",
	            "number": 4281,
	            "title": "Add test reproducing #4063",
	            "url": "https://github.com/apollographql/apollo-server/pull/4281",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDM1MDQ4MTc3",
	            "number": 4269,
	            "title": "chore(deps): Updating hapi package and ts types for apollo-server-hapi",
	            "url": "https://github.com/apollographql/apollo-server/pull/4269",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDM0MTAwMTMy",
	            "number": 4257,
	            "title": "fix(4256): Fix type mismatch between apollo-datasource-rest & apollo-server-caching",
	            "url": "https://github.com/apollographql/apollo-server/pull/4257",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDMzOTk5NzM3",
	            "number": 4254,
	            "title": "chore(deps): update dependency accept to @hapi/accept and boom to @ha…",
	            "url": "https://github.com/apollographql/apollo-server/pull/4254",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDMzMzY5MDIx",
	            "number": 4239,
	            "title": "fix(gateway): maintain immutability when merging selection sets",
	            "url": "https://github.com/apollographql/apollo-server/pull/4239",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDMxMzEyNTQ0",
	            "number": 4222,
	            "title": "Feature: allow `dataSources` creator function to be async",
	            "url": "https://github.com/apollographql/apollo-server/pull/4222",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDI3ODgyNzg1",
	            "number": 4203,
	            "title": "Add support for passing options to http-cache-semantics in apollo-datasource-rest",
	            "url": "https://github.com/apollographql/apollo-server/pull/4203",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDI3NzQ1NDA4",
	            "number": 4201,
	            "title": "Fix federation Query/Mutation extends documentation",
	            "url": "https://github.com/apollographql/apollo-server/pull/4201",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDI3NTI3ODIw",
	            "number": 4200,
	            "title": "Fix server type check",
	            "url": "https://github.com/apollographql/apollo-server/pull/4200",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDI0NzU0NTE0",
	            "number": 4176,
	            "title": "Change keyValueCache property of HTTPCache from private to protected.",
	            "url": "https://github.com/apollographql/apollo-server/pull/4176",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDI0MTk3NDYw",
	            "number": 4172,
	            "title": "Export ContextFunction from apollo-server-express",
	            "url": "https://github.com/apollographql/apollo-server/pull/4172",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDIzNzAwODM5",
	            "number": 4167,
	            "title": "Custom execute and subscribe functions",
	            "url": "https://github.com/apollographql/apollo-server/pull/4167",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDIzNjU2ODIy",
	            "number": 4166,
	            "title": ":sparkles: execute operation with external context",
	            "url": "https://github.com/apollographql/apollo-server/pull/4166",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDIzNDgzMTM5",
	            "number": 4165,
	            "title": "Return 400, not 500, when POST body missing",
	            "url": "https://github.com/apollographql/apollo-server/pull/4165",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDIyODczMTU1",
	            "number": 4162,
	            "title": "Improve defaultMaxAge setting",
	            "url": "https://github.com/apollographql/apollo-server/pull/4162",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDIyMzEwNjcz",
	            "number": 4157,
	            "title": "fix(gateway): prevent null fields from expanding into objects when passed to resolvers",
	            "url": "https://github.com/apollographql/apollo-server/pull/4157",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDIyMDY0MzAz",
	            "number": 4148,
	            "title": "Federation: Insert empty Subscription type definition",
	            "url": "https://github.com/apollographql/apollo-server/pull/4148",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDE3NzIzMDQ2",
	            "number": 4109,
	            "title": "Mention that formatError masks without debug: true",
	            "url": "https://github.com/apollographql/apollo-server/pull/4109",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDE0NzE4Njk3",
	            "number": 4078,
	            "title": "Change status code to 400 when POST body is missing",
	            "url": "https://github.com/apollographql/apollo-server/pull/4078",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDExNjI2MjMx",
	            "number": 4051,
	            "title": "feat(core): refactored requestDidStart event to be possibly asynchronous",
	            "url": "https://github.com/apollographql/apollo-server/pull/4051",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDExNTExMzUw",
	            "number": 4050,
	            "title": "feat(core): refactored invokeDidStartHook to accept async resolvig hooks/events",
	            "url": "https://github.com/apollographql/apollo-server/pull/4050",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDExNTAyNjU1",
	            "number": 4049,
	            "title": "feat(core): added graceful error handler for validationDidStart hook",
	            "url": "https://github.com/apollographql/apollo-server/pull/4049",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDA5NTY1ODY4",
	            "number": 4039,
	            "title": "Upgrade graphql-upload to support Node.js v14",
	            "url": "https://github.com/apollographql/apollo-server/pull/4039",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDA0ODQ2ODkw",
	            "number": 4000,
	            "title": "Update Documentation Metric for federation to use ENGINE_SCHEMA_TAG env ",
	            "url": "https://github.com/apollographql/apollo-server/pull/4000",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0NDAzODY2NDcy",
	            "number": 3987,
	            "title": "Allow to use @provide with interfaces & Allow to use @extend one fields defined in service different then base",
	            "url": "https://github.com/apollographql/apollo-server/pull/3987",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzk3MjQ4NDY3",
	            "number": 3940,
	            "title": "Rest datasource trace protected",
	            "url": "https://github.com/apollographql/apollo-server/pull/3940",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzk2MzU5MzU1",
	            "number": 3933,
	            "title": "Add response headers to error object.",
	            "url": "https://github.com/apollographql/apollo-server/pull/3933",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzkyMDc3MDEx",
	            "number": 3914,
	            "title": "Handle external types that do not exist gracefully",
	            "url": "https://github.com/apollographql/apollo-server/pull/3914",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzkwMzAwNDgy",
	            "number": 3897,
	            "title": "Require missing defaultFieldResolver in docs",
	            "url": "https://github.com/apollographql/apollo-server/pull/3897",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg5NTExMzE5",
	            "number": 3895,
	            "title": "apollo-server-fastify's context function now receives reply object.",
	            "url": "https://github.com/apollographql/apollo-server/pull/3895",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg3ODkwNjcy",
	            "number": 3888,
	            "title": "federation spec: clarify handling of types with all fields @external",
	            "url": "https://github.com/apollographql/apollo-server/pull/3888",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg2NDg4MTM5",
	            "number": 3880,
	            "title": "fixes subscriptions install bug",
	            "url": "https://github.com/apollographql/apollo-server/pull/3880",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg2MjQ1MDU1",
	            "number": 3878,
	            "title": "[RESTDataSource] refactor to allow overridding the Error class",
	            "url": "https://github.com/apollographql/apollo-server/pull/3878",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg1MzI4NDQ2",
	            "number": 3875,
	            "title": "feat: support customizing micro request body parsing options",
	            "url": "https://github.com/apollographql/apollo-server/pull/3875",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg0Nzg1MTQ3",
	            "number": 3866,
	            "title": "full-response-cache: include `extensions` in cache.",
	            "url": "https://github.com/apollographql/apollo-server/pull/3866",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzg0MjQ1MzM2",
	            "number": 3862,
	            "title": "Add missing options for subscription server",
	            "url": "https://github.com/apollographql/apollo-server/pull/3862",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzgzMTEzNjM5",
	            "number": 3854,
	            "title": "docs: Deploying with Azure Functions / uri correction",
	            "url": "https://github.com/apollographql/apollo-server/pull/3854",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzgyMDg4NTc4",
	            "number": 3850,
	            "title": "Fix file upload bug with apollo-server-koa (#2843)",
	            "url": "https://github.com/apollographql/apollo-server/pull/3850",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzc2NzM1MDA3",
	            "number": 3806,
	            "title": "fix(RESTDataSource): avoid reference errors when using memoized results ",
	            "url": "https://github.com/apollographql/apollo-server/pull/3806",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mzc1ODQxMjY1",
	            "number": 3799,
	            "title": "Implement internal directive",
	            "url": "https://github.com/apollographql/apollo-server/pull/3799",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzcyMzg2MjQ5",
	            "number": 3766,
	            "title": "feat(RESTDataSource): add `cacheKey` override",
	            "url": "https://github.com/apollographql/apollo-server/pull/3766",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzcyMzAyODQ1",
	            "number": 3765,
	            "title": "feat: Add support for including excepting `SourceCharacter` to `printDescription`",
	            "url": "https://github.com/apollographql/apollo-server/pull/3765",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzcwMzgzODM2",
	            "number": 3754,
	            "title": "Allow custom fetch instance to be passed into remote gateway datasource",
	            "url": "https://github.com/apollographql/apollo-server/pull/3754",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzY3MTI3Njcx",
	            "number": 3723,
	            "title": "Update graphql-upload to v10.0.0",
	            "url": "https://github.com/apollographql/apollo-server/pull/3723",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzY0NTIzMzQ4",
	            "number": 3690,
	            "title": "Add new pubsub implementation",
	            "url": "https://github.com/apollographql/apollo-server/pull/3690",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzYzNzg3NjQ3",
	            "number": 3681,
	            "title": "Add sampleRate option to limit data volume",
	            "url": "https://github.com/apollographql/apollo-server/pull/3681",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzYwOTg1MDg3",
	            "number": 3660,
	            "title": "feat(apollo-gateway): opentracing",
	            "url": "https://github.com/apollographql/apollo-server/pull/3660",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzU2ODk2MjYw",
	            "number": 3636,
	            "title": "Respect is base64 encoded",
	            "url": "https://github.com/apollographql/apollo-server/pull/3636",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzUzNjcwNDY1",
	            "number": 3615,
	            "title": "Fix security vulnerabilities in documentation",
	            "url": "https://github.com/apollographql/apollo-server/pull/3615",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzUzMzYwNDY4",
	            "number": 3610,
	            "title": "federation: Include originalError in downstreamServiceError",
	            "url": "https://github.com/apollographql/apollo-server/pull/3610",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzUwMjk0Njg1",
	            "number": 3587,
	            "title": "Narrow willSendRequest's parameter type",
	            "url": "https://github.com/apollographql/apollo-server/pull/3587",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzQ3NTYwOTU1",
	            "number": 3573,
	            "title": "added a null check for serialized value as well",
	            "url": "https://github.com/apollographql/apollo-server/pull/3573",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzQyNTY1Njg4",
	            "number": 3528,
	            "title": "chore(package) update fastify",
	            "url": "https://github.com/apollographql/apollo-server/pull/3528",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzQxNTk0MTY3",
	            "number": 3514,
	            "title": "Add readme for apollo-server-cloudflare",
	            "url": "https://github.com/apollographql/apollo-server/pull/3514",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzM0ODk5MTM0",
	            "number": 3460,
	            "title": "Adds cors options available to configure",
	            "url": "https://github.com/apollographql/apollo-server/pull/3460",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzMyNDczNTkz",
	            "number": 3441,
	            "title": "changed accept package to @hapi/accept in apollo-server-micro",
	            "url": "https://github.com/apollographql/apollo-server/pull/3441",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzIyNDc4MTEz",
	            "number": 3353,
	            "title": "add next function to support koa middleware processing.",
	            "url": "https://github.com/apollographql/apollo-server/pull/3353",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzE3NzcyNjQ2",
	            "number": 3296,
	            "title": "Add subscription support for testing client & improved typings",
	            "url": "https://github.com/apollographql/apollo-server/pull/3296",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzE1ODIyNDg1",
	            "number": 3273,
	            "title": "Upgrade hapi package",
	            "url": "https://github.com/apollographql/apollo-server/pull/3273",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzE1MzE1MzU1",
	            "number": 3268,
	            "title": "Bug/#3267 hapi upload",
	            "url": "https://github.com/apollographql/apollo-server/pull/3268",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzEzMzU0NDc1",
	            "number": 3244,
	            "title": "Allow to use @provide with interfaces",
	            "url": "https://github.com/apollographql/apollo-server/pull/3244",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzA2MzM1Nzk2",
	            "number": 3148,
	            "title": "Fix stack trace when using apollo-server-testing",
	            "url": "https://github.com/apollographql/apollo-server/pull/3148",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MzAyNTQwODU4",
	            "number": 3108,
	            "title": "Updating the lambda handler to handle invalid JSON body",
	            "url": "https://github.com/apollographql/apollo-server/pull/3108",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mjk1NjQ0NDE4",
	            "number": 3017,
	            "title": "[DataSource] Add missing class property context",
	            "url": "https://github.com/apollographql/apollo-server/pull/3017",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0Mjk1NTMwNTM1",
	            "number": 3013,
	            "title": "Build Federated Schemas From Existing Schema",
	            "url": "https://github.com/apollographql/apollo-server/pull/3013",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjkxMjYxMTUy",
	            "number": 2909,
	            "title": "feat(lambda): support async handler functionality",
	            "url": "https://github.com/apollographql/apollo-server/pull/2909",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjcyODgyMTM4",
	            "number": 2601,
	            "title": "Make ApolloServerBase#{context,subscriptionServer} protected, not pri…",
	            "url": "https://github.com/apollographql/apollo-server/pull/2601",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjcyMjQ3Njcx",
	            "number": 2600,
	            "title": "apollo-server-lambda/src/lambdaApollo no longer errors when event.isBase64Encoded",
	            "url": "https://github.com/apollographql/apollo-server/pull/2600",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjcwNjEzMzEw",
	            "number": 2576,
	            "title": "docs how to implement modules",
	            "url": "https://github.com/apollographql/apollo-server/pull/2576",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjY0MDI0OTgy",
	            "number": 2489,
	            "title": "Add support for human-readable maxAge",
	            "url": "https://github.com/apollographql/apollo-server/pull/2489",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjU3Mzk2MTU2",
	            "number": 2391,
	            "title": "Fastify: update to version 2",
	            "url": "https://github.com/apollographql/apollo-server/pull/2391",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjU1NjU2OTY3",
	            "number": 2367,
	            "title": "[WIP] uWebSockets Integration",
	            "url": "https://github.com/apollographql/apollo-server/pull/2367",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjUzNzI5MTMz",
	            "number": 2333,
	            "title": "Feature: Add HTTP2 support",
	            "url": "https://github.com/apollographql/apollo-server/pull/2333",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjQ4NDM0MTI1",
	            "number": 2241,
	            "title": "fix: Use event.requestContext.path on lambda playground when available",
	            "url": "https://github.com/apollographql/apollo-server/pull/2241",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjQ3OTcxMjMy",
	            "number": 2232,
	            "title": "Re-add dynamic validation by request",
	            "url": "https://github.com/apollographql/apollo-server/pull/2232",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjM1MjEwMjU4",
	            "number": 2058,
	            "title": "Docs – Deployment to Firebase Cloud Functions",
	            "url": "https://github.com/apollographql/apollo-server/pull/2058",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjMxMzM3MTY1",
	            "number": 1974,
	            "title": "Support playgroundPath",
	            "url": "https://github.com/apollographql/apollo-server/pull/1974",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjIxOTIzOTU2",
	            "number": 1799,
	            "title": "Added GraphQL Middleware support",
	            "url": "https://github.com/apollographql/apollo-server/pull/1799",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjE4MzMzMTI1",
	            "number": 1734,
	            "title": "Getting graphql-playground-react to work when offline",
	            "url": "https://github.com/apollographql/apollo-server/pull/1734",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjE2MDg1MjU0",
	            "number": 1683,
	            "title": "Allow willResolveField's handlers to modify the result",
	            "url": "https://github.com/apollographql/apollo-server/pull/1683",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        },
	        {
	          "node": {
	            "id": "MDExOlB1bGxSZXF1ZXN0MjA2MjQ0ODEx",
	            "number": 1490,
	            "title": "[WIP] Add support for `Server Timing` headers",
	            "url": "https://github.com/apollographql/apollo-server/pull/1490",
	            "state": "CLOSED",
	            "baseRefName": "master"
	          }
	        }
	      ]
	    }
	  }
	}
	```
	
	</details>

2. Here's what I cobbled together to undo it — it requires [`jq`](https://stedolan.github.io/jq/) and replacing `MY_GITHUB_TOKEN`:

	```sh
	curl --request POST \
	  --url https://api.github.com/graphql \
	  --silent -S \
	  --header 'authorization: token MY_GITHUB_TOKEN' \
	  --header 'content-type: application/json' \
	  --data '{
	            "query": "query ($search: String!) { prSearch: search(type: ISSUE, first: 100, query: $search) { issueCount edges { node { ... on PullRequest { id number title url state baseRefName } } } } }",
	            "variables": {
	              "search": "repo:apollographql/apollo-server is:pr closed:2020-06-24T21:10:00+03:00..2020-06-24T21:20:00+03:00 is:closed -author:app/renovate"
	            }
	          }' |
	  jq -crM '.data.prSearch.edges[].node.id | { query: "mutation ($pullRequestId: ID! $newBranchName: String!) { reopenPullRequest(input: { pullRequestId: $pullRequestId }) { pullRequest { id number title baseRefName url } } updatePullRequest(input: { pullRequestId: $pullRequestId baseRefName: $newBranchName }) { pullRequest { id number title baseRefName url } } }", variables: { pullRequestId: ., newBranchName: "main" } }' |
	  while read -r line; do
	       echo "$line" |
	         curl --request POST \
	            --url https://api.github.com/graphql \
	            --silent -S \
	            --header 'content-type: application/json' \
	            --header 'authorization: token MY_GITHUB_TOKEN' \
	            --data @- |
	              jq .
	  done
	```
